### PR TITLE
Adds scripting guide doc

### LINKED ISSUE #9317: [Docs] Add requirement to always invoke powershell with -NoProfile option to coding guidelines
I have encountered several issues with powershell invocations in coreclr build, CI and test code which were caused by powershell invocation without using -NoProfile option which prevents powershell profile script execution on startup.

Unlike Linux shells powershell when invoked by default executes profile scripts even on unixes. This overlooked feature may cause side effects which may affect executing code. To guard against this problem it should be a requirement to always invoke powershell with -NoProfile option.

See dotnet/coreclr#15164

### PR DESCRIPTION
**Pull request by marcusturewicz**
Scripting guide for Shell, PowerShell and CMD files for styling or other guidance.

I've add a some general guidance at the opening, which follows the coding guidelines regarding non-C# code. I've then added a section about always including `-NoProfile` to PowerShell commands. If anything in the future pops up regarding Shell or CMD scripts, relevant sections can also be added.

Fixes #9317

### CODE REVIEW

File: docs/coding-guidelines/coding-style.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 C# Coding Style
 ===============
 
-For C++ files (*.cpp and *.h), we use clang-format (version 3.6+) to ensure code styling. After changing any Cpp or H file and before merging, src/Native/format-code.sh must be run. This script will ensure that all native code files adhere to the coding style guidelines.
+For C++ files (*.cpp and *.h), we use clang-format (version 3.6+) to ensure code styling. After changing any Cpp or H file and before merging, src/Native/format-code.sh must be run. This script will ensure that all native code files adhere to the coding style guidelines. For script files (*.sh, *.ps1 and *.cmd), see the [Scripting Guide](scripting-guide.md).
 
 For non code files (xml, etc), our current best guidance is consistency. When editing files, keep new code and changes consistent with the style in the files. For new files, it should conform to the style for that component. If there is a completely new component, anything that is reasonably broadly accepted is fine.
```
Comment (jkotas) [inline docs/coding-guidelines/coding-style.md:6]:
```suggestion
For other types of files (xml, bat, sh, etc), our current best guidance is consistency. When editing files, keep new code and changes consistent with the style in the files. For new files, it should conform to the style for that component. If there is a completely new component, anything that is reasonably broadly accepted is fine.
```

Comment (jkotas) [reply docs/coding-guidelines/coding-style.md:6]: Better than duplicating the paragraph

File: docs/coding-guidelines/scripting-guide.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Scripting Guide
+===============
+
+Generally, for script files (.sh, .ps1 & .cmd), our current best guidance is consistency. When editing files, keep new code and changes consistent with the style in the files. For new files, it should conform to the style for that component. If there is a completely new component, anything that is reasonably broadly accepted is fine.
+
+PowerShell
+-----------------
+
+### -NoProfile
+Unlike Unix or Linux based shells, when PowerShell is invoked by default it executes [profile scripts](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_profiles), even on Unix or Linux systems. This feature may cause side effects which may affect executing code. To guard against this, it is a requirement to always invoke PowerShell with the `-NoProfile` option. For example, the following PowerShell command in valid:
```
Comment (jkotas) [inline docs/coding-guidelines/scripting-guide.md:10]: This is just one of the many best practices for writing and using PowerShell scripts. I do not think we want to maintain list of best practices for writing shell scripts in this repo, the shell scripts are just a small fraction of it.

Can we just link to a good list of Powershell best practices from the other doc?

Comment (marcusturewicz) [reply docs/coding-guidelines/scripting-guide.md]: Have updated per your suggestions. Have linked to the [Scripting devblog](https://devblogs.microsoft.com/scripting) which, although slightly hard to find, there is an article about using [-NoProfile](https://devblogs.microsoft.com/scripting/powertip-launch-powershell-without-a-profile). This was the best resources I could find regarding "official" PowerShell best practices.

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**